lib/puppet/provider/templates/invoke_dsc_resource.ps1.erb

```diff
@@ -31,13 +34,25 @@ $invokeParams = @{
 <% end -%>
 }
 
-$result = Invoke-DscResource @invokeParams
+try{
+    $result = Invoke-DscResource @invokeParams
+}catch{
+  $exc = $_
+  $data = [PSCustomObject]@{
+    "error" = "execfailure"
+    "exception" = $_.Exception.Message
```

Comment:
Message is probably sufficient ... but I have a helper I've written in the past for better parsing of the exception object (SQL stuff won't be relevant of course) - https://github.com/Iristyle/Midori/blob/master/tools/Powershell-Contrib.psm1#L270-L392